contracts/nft-querier/src/msg.rs

```diff
@@ -0,0 +1,34 @@
+use crate::state::Config;
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::Addr;
+use cw_ownable::{cw_ownable_execute, cw_ownable_query};
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub factory_contract: Addr,
```

Comment:
Never accept `Addr` from external sources. There are a few basic rules to follow:
- if you receive an address from an external source, it must be of type `String`. Examples of external sources: `ExecuteMsg`, `QueryMsg`, `InstantiateMsg`, etc.;
- once you receive an address from an external source, run `deps.api.addr_validate()` on it to convert it into `cosmwasm_std::Addr`;
- if you need to save an address to persistent storage, save it as `Addr`.

```suggestion
    pub factory_contract: String,
```